src/snowflake/cli/plugins/snowpark/package/anaconda.py

```diff
@@ -57,11 +64,22 @@ def parse_anaconda_packages(self, packages: List[Requirement]) -> SplitRequireme
         for package in packages:
             if package.extras:
                 result.other.append(package)
-            elif self.is_package_available(package):
+            elif self.is_package_available(
+                package, skip_version_check=skip_version_check
+            ):
                 result.snowflake.append(package)
             else:
                 log.info(
                     "'%s' not found in Snowflake Anaconda channel...", package.name
                 )
                 result.other.append(package)
         return result
+
+
+def get_anaconda_from_snowflake() -> AnacondaChannel:
+    try:
+        return AnacondaChannel.from_snowflake()
+    except HTTPError as err:
+        raise ClickException(
+            f"Accessing Snowflake Anaconda channel failed. Reason {err}"
+        )
```

Comment:
Consider moving this to `from_snowflake`, I'm not sure if this need a factory method

Reply:
fixed